### DISCUSSION TITLE
关于目前最新的字幕识别OCR模型

### DISCUSSION THREAD

**Discussion by joseph16388** [SWHL/RapidVideOCR#66] [Q&A] [answered]
你好！
1、目前rapidocr_onnxruntime调用的最新的模型是ch_ppocr_mobile_v2.0_cls_infer、ch_PP-OCRv4_det_infer、ch_PP-OCRv4_rec_infer吗？
如果不在乎识别速度，有更大更好更新的模型替换这几个模型吗？
2、如何微调这几个模型，有详细的教程吗？
谢谢！

**Comment by SWHL** [comment]
QA1: 这个可以用[link](https://huggingface.co/SWHL/RapidOCR/tree/main/PP-OCRv4)下server的模型来选择性替换这几个轻量模型。具体替换教程参见：[docs](https://rapidai.github.io/RapidOCRDocs/blog/2023/10/17/%E5%A6%82%E4%BD%95%E6%9B%B4%E6%8D%A2%E5%85%B6%E4%BB%96%E6%A3%80%E6%B5%8B%E5%92%8C%E8%AF%86%E5%88%AB%E6%A8%A1%E5%9E%8B/#3)

QA2: 目前RapidOCR不支持微调模型，想要微调的话， 请移步[PaddleOCR](https://github.com/PaddlePaddle/PaddleOCR)
RapidOCR和PaddleOCR的关系如下图：
<img width="1188" alt="image" src="https://github.com/user-attachments/assets/026b989a-a815-4ac9-aee4-87d9dc7e0346">

**Reply by joseph16388** [reply]
你好！我目前用的是：
vsf_ocr = RapidVideoSubFinderOCR(
            vsf_exe_path=vsf_exe_path,
            is_concat_rec=True,
            create_cleared_text_images=False,
            open_video_opencv=open_video_opencv,
            open_video_ffmpeg=open_video_ffmpeg,
            use_cuda=True,
            top_video_image_percent_end=top_video_image_percent_end,
            bottom_video_image_percent_end=bottom_video_image_percent_end,
            left_video_image_percent_end=left_video_image_percent_end,
            right_video_image_percent_end=right_video_image_percent_end
        )
RapidVideoSubFinderOCR这个类，但是这个类没有读取Det和Rec模型本地路径的接口。。

**Reply by SWHL** [reply]
咦。。。的确没有这个接口，我有空更新一下吧。着急使用的话，在pip安装之后，直接在这个[位置](https://github.com/SWHL/RapidVideOCR/blob/296bb37e69d2d2e2135eade73eb0cd803f68629a/rapid_videocr/main.py#L70)修改一下源码吧

**Reply by joseph16388** [reply]
> 咦。。。的确没有这个接口，我有空更新一下吧。着急使用的话，在pip安装之后，直接在这个[位置](https://github.com/SWHL/RapidVideOCR/blob/296bb37e69d2d2e2135eade73eb0cd803f68629a/rapid_videocr/main.py#L70)修改一下源码吧

好的，感谢！

**Reply by SWHL** [reply, marked as answer]
已经在rapidvideocr>=2.2.14中支持，请再次安装尝试

**Reply by joseph16388** [reply]
> 已经在rapidvideocr>=2.2.14中支持，请再次安装尝试

好的,非常感谢！

**Reply by joseph16388** [reply]
你好！不过server版的OCR时间实在太长了，有没有提速或者使用其他模型折中的方案呢？
